server/src/config/connectionStringBuilder.ts

```diff
@@ -19,15 +11,23 @@ export class MongoConnection {
     this.databaseName = this.conn.databaseName;
     this.reconnectInterval = 10000;
   }
-  getConnection = () => {
+
+  getConnection(): {
+    username: string;
+    passwd: string;
+    host: string;
+    port: string;
+    databaseName: string;
+    connectionOptions: string;
+  } {
     const envConnectionOptions = process.env.MONGO_CONNECTION_OPTIONS;
     const defaultConnectionOptions = 'maxPoolSize=20&w=majority';
-    const conn = {
-      username: process.env.MONGO_USERNAME,
-      passwd: process.env.MONGO_PASSOWORD,
-      host: process.env.MONGO_HOST,
-      port: process.env.MONGO_PORT,
-      databaseName: process.env.MONGO_DATABASE,
+    const conn: Connection = {
+      username: process.env.MONGO_USERNAME || '',
+      passwd: process.env.MONGO_PASSOWORD || '',
+      host: process.env.MONGO_HOST || '',
+      port: process.env.MONGO_PORT || '',
+      databaseName: process.env.MONGO_DATABASE || '',
```

Comment:
In both options you have to write guards, which is a short expression that checks a type. Two lines below there is check for each parameter which identifies if all of the parameters were given, if not if throws an error with "parameters X, Y are missing"

If we want to use approach with "string | undefinied" then we have to add 2 guards, one to check if type is not undefinied (one on the bottom of main function and second in 'createUri').  You cannot just define this as "string" and then leave it undefinied if possible, then TS wont compile.

However I will change it to 'undefinied'. It will be more semantic that if there is error, the value was not given. It is more clear that: "the value was not given or was given empty".